lib/byron/src/Cardano/Wallet/Byron.hs

```diff
@@ -224,57 +198,45 @@ serveWallet
         withNetworkLayer networkTracer gbp socketPath vData $ \nl -> do
             withWalletNtpClient io ntpClientTracer $ \ntpClient -> do
                 let pm = fromNetworkMagic $ networkMagic $ fst vData
-                randomApi  <- apiLayer (newTransactionLayer proxy pm) nl
-                icarusApi  <- apiLayer (newTransactionLayer proxy pm) nl
-                startServer proxy socket randomApi icarusApi ntpClient
+                randomApi  <- apiLayer (newTransactionLayer n pm) nl (byronScheme n)
+                icarusApi  <- apiLayer (newTransactionLayer n pm) nl (icarusScheme n)
+                startServer socket randomApi icarusApi ntpClient
                 pure ExitSuccess
 
-    networkDiscriminantValFromProxy
-        :: forall n. (NetworkDiscriminantVal n)
-        => Proxy n
-        -> Text
-    networkDiscriminantValFromProxy _ =
-        networkDiscriminantVal @n
-
     startServer
-        :: forall n.
-            ( PaymentAddress n IcarusKey
-            , PaymentAddress n ByronKey
-            , DecodeAddress n
```

Comment:
We were forced to explicitly specify these constraints because of the `n` parameter.

I think they are implementation details, and being able to remove them here is a good thing.